File: .opencode/skills/mysql/scripts/update.js
#!/usr/bin/env node
/**
 * MySQL Update - Node.js 腳本
 * 更新 MySQL 資料表中的記錄
 */

const mysql = require('mysql2/promise');

async function getConnection() {
  try {
    const conn = await mysql.createConnection({
      host: process.env.MYSQL_HOST || '165.154.226.78',
      port: parseInt(process.env.MYSQL_PORT) || 3306,
      user: process.env.MYSQL_USER || 'n8n',
      password: process.env.MYSQL_PASSWORD || '!!asshole!!asshole',
      database: process.env.MYSQL_DATABASE || 'infoCollection'
    });
    return conn;
  } catch (error) {
    console.error(`資料庫連線失敗: ${error.message}`);
    process.exit(1);
  }
}

async function updateRecord(table, recordId, data) {
  const conn = await getConnection();

  try {
    const setClause = Object.keys(data).map(k => `${k} = ?`).join(', ');
    const values = [...Object.values(data), recordId];

    const sql = `UPDATE ${table} SET ${setClause} WHERE id = ?`;
    const [result] = await conn.execute(sql, values);

    return result.affectedRows;
  } catch (error) {
    console.error(`更新失敗: ${error.message}`);
    process.exit(1);
  } finally {
    await conn.end();
  }
}

async function main() {
  if (process.argv.length < 5) {
    console.log('使用方法: node update.js <table> <id> <data>');
    console.log('範例:');
    console.log('  node update.js users 5 \'{"name":"Updated Name","age":26}\'');
    console.log('  node update.js products 10 \'{"price":89.99}\'');
    process.exit(1);
  }

  const table = process.argv[2];
  const recordId = parseInt(process.argv[3], 10);

  if (isNaN(recordId)) {
    console.error('錯誤: id 必須是整數');
    process.exit(1);
  }

  let data;

  try {
    data = JSON.parse(process.argv[4]);
  } catch (error) {
    console.error(`JSON 解析失敗: ${error.message}`);
    process.exit(1);
  }

  if (typeof data !== 'object' || data === null || Array.isArray(data)) {
    console.error('錯誤: data 必須是 JSON 物件格式');
    process.exit(1);
  }

  const affectedRows = await updateRecord(table, recordId, data);
  console.log(`成功更新記錄，影響 ${affectedRows} 行`);
}

main().catch(error => {
  console.error('未預期的錯誤:', error);
  process.exit(1);
});

File: mysql/scripts/update.py
#!/usr/bin/env python3
"""
MySQL Update - Python 腳本
更新 MySQL 資料表中的記錄
"""

import os
import sys
import json
import mysql.connector

def get_connection():
    """建立資料庫連線"""
    try:
        conn = mysql.connector.connect(
            host=os.environ.get('MYSQL_HOST', '192.168.1.159'),
            port=int(os.environ.get('MYSQL_PORT', 3306)),
            user=os.environ.get('MYSQL_USER', 'n8n'),
            password=os.environ.get('MYSQL_PASSWORD', '!!asshole!!asshole'),
            database=os.environ.get('MYSQL_DATABASE', 'infoCollection')
        )
        return conn
    except mysql.connector.Error as e:
        print(f"資料庫連線失敗: {e}", file=sys.stderr)
        sys.exit(1)

def update_record(table, record_id, data):
    """
    更新資料表中的記錄

    Args:
        table: 資料表名稱
        record_id: 記錄 ID
        data: 要更新的資料（字典格式）

    Returns:
        受影響的行數
    """
    conn = get_connection()
    cursor = conn.cursor(dictionary=True)

    try:
        set_clause = ', '.join([f"{k} = %s" for k in data.keys()])
        sql = f"UPDATE {table} SET {set_clause} WHERE id = %s"

        values = tuple(data.values()) + (record_id,)
        cursor.execute(sql, values)
        conn.commit()

        affected_rows = cursor.rowcount
        return affected_rows
    except mysql.connector.Error as e:
        conn.rollback()
        print(f"更新失敗: {e}", file=sys.stderr)
        sys.exit(1)
    finally:
        cursor.close()
        conn.close()

def main():
    if len(sys.argv) < 4:
        print("使用方法: python3 update.py <table> <id> <data>")
        print("範例:")
        print('  python3 update.py users 5 \'{"name":"Updated Name","age":26}\'')
        print('  python3 update.py products 10 \'{"price":89.99}\'')
        sys.exit(1)

    table = sys.argv[1]
    try:
        record_id = int(sys.argv[2])
    except ValueError:
        print("錯誤: id 必須是整數", file=sys.stderr)
        sys.exit(1)

    try:
        data = json.loads(sys.argv[3])
    except json.JSONDecodeError as e:
        print(f"JSON 解析失敗: {e}", file=sys.stderr)
        sys.exit(1)

    if not isinstance(data, dict):
        print("錯誤: data 必須是 JSON 物件格式", file=sys.stderr)
        sys.exit(1)

    affected_rows = update_record(table, record_id, data)
    print(f"成功更新記錄，影響 {affected_rows} 行")

if __name__ == "__main__":
    main()
